src/versions.rs

```diff
@@ -0,0 +1,39 @@
+use std::str::FromStr;
+
+#[derive(Debug, Clone, PartialEq)]
+// major, minor, patch
+pub struct Version(pub u16, pub u16, pub u16);
+
+impl FromStr for Version {
+    type Err = String;
+
+    /// Parse a `Version` from a string.
+    fn from_str(s: &str) -> Result<Version, String> {
+        let mut parts = s.splitn(4, '.')
+            .map(|s| s.parse())
+            .take_while(Result::is_ok)
+            .map(Result::unwrap);
+        let version: (u16, u16, u16) = (parts.next().unwrap_or(0),
```

Comment:
Now you can stick the name `Version` in here and avoid enumerating the fields below.